src/sysupdate/sysupdate-resource.c

```diff
@@ -527,13 +527,25 @@ int resource_load_instances(Resource *rr, bool verify, Hashmap **web_cache) {
         return 0;
 }
 
+static int instance_version_match(Instance *const*a, Instance *const*b) {
+        assert(a);
+        assert(b);
+        assert(*a);
+        assert(*b);
+        assert((*a)->metadata.version);
```

Comment:
This will abort if a version is `0`, is that intended?

Reply:
the version is a char*, not a number